src/io.jl

```diff
@@ -18,341 +18,92 @@ function ioraise(e)
+        throw(ArgumentError("unrecognized whence=$n argument to seek"))
     end
-    return convert(PyPtr, C_NULL)
+    return position(io)
 end
 
-##########################################################################
-# TODO: support other Python interfaces (e.g. TextIO) when possible?
 
 ##########################################################################
 
-# make method tables for text/binary IO objects (called in __io__)
-make_io_methods(text::Bool) =
-    PyMethodDef[
-                PyMethodDef("close", jl_IO_close, METH_NOARGS),
-                PyMethodDef("fileno", jl_IO_fileno, METH_NOARGS),
-                PyMethodDef("flush", jl_IO_flush, METH_NOARGS),
-                PyMethodDef("isatty", jl_IO_isatty, METH_NOARGS),
-                PyMethodDef("readable", jl_IO_readable, METH_NOARGS),
-                PyMethodDef("writable", jl_IO_writable, METH_NOARGS),
-                PyMethodDef("readline", jl_IO_readline, METH_VARARGS),
-                PyMethodDef("readlines", jl_IO_readlines, METH_VARARGS),
-                PyMethodDef("seek", jl_IO_seek, METH_VARARGS),
-                PyMethodDef("seekable", jl_IO_seekable, METH_NOARGS),
-                PyMethodDef("tell", jl_IO_tell, METH_NOARGS),
-                PyMethodDef("writelines", jl_IO_writelines, METH_O),
-                PyMethodDef("read", text ? jl_IO_read_text : jl_IO_read, METH_VARARGS),
-                PyMethodDef("readall", text ? jl_IO_readall_text : jl_IO_readall, METH_NOARGS),
-                PyMethodDef("readinto", jl_IO_readinto, METH_O),
-                PyMethodDef("write", jl_IO_write, METH_O),
-                PyMethodDef() # sentinel
-    ]
-
-
-function pyio_repr(o::PyPtr)
-    o = PyObject(try string("<PyCall.io ",unsafe_pyjlwrap_to_objref(o),">")
-                 catch "<PyCall.io NULL>"; end)
-    oret = o.o
-    o.o = convert(PyPtr, C_NULL) # don't decref
-    return oret
-end
+pyio_jl(self::PyObject) = unsafe_pyjlwrap_to_objref(self["io"].o)::IO
 
 pyio_initialized = false
 function pyio_initialize()
     global pyio_initialized
     if !pyio_initialized::Bool
-        global const jl_IOType =
-            pyjlwrap_type("PyCall.jl_IO",
-                          t -> begin
-                              t.tp_getattro = @pyglobal(:PyObject_GenericGetAttr)
-                              t.tp_methods = pointer(jl_IO_methods)
-                              t.tp_getset = pointer(jl_IO_getset)
-                              t.tp_repr = pyio_repr_ptr
-                          end)
-        global const jl_TextIOType =
-            pyjlwrap_type("PyCall.jl_TextIO",
-                          t -> begin
-                              t.tp_getattro = @pyglobal(:PyObject_GenericGetAttr)
-                              t.tp_methods = pointer(jl_TextIO_methods)
-                              t.tp_getset = pointer(jl_IO_getset)
-                              t.tp_repr = pyio_repr_ptr
-                          end)
+        @eval @pydef type PyIO
```

Comment:
It would be nice to have a lower-level macro so that we don't have to use `@eval`, i.e. a `@pydef` macro that simply returns the Python type object without defining any Julia constant.   In the absence of a better name, maybe `@pydef_`.   Then we can do:
```jl
const PyIO = PyNULL()
function pyio_initialize()
    if PyIO.o == C_NULL
        copy!(PyIO, @pydef_ .....)
end
```
(I'd like to move away from all of these `global const Foo = ....` definitions that I currently have, since they screw up type inference in precompiled code.)